ir/base/smart.go

```diff
@@ -0,0 +1,93 @@
+package base
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/libp2p/go-smart-record/ir"
+)
+
+//NOTE: Including everything in the same file for discussion purposes.
+// This should be organized in its corresponding files when implementing it.
+type SmartNode interface {
+	ir.Node
+	Disassemble() ir.Node // returns only syntactic nodes
+}
+
+// SmartString is the smart node for the String type.
+// The difference is that additional metadata is included in the assembly process
+type SmartString struct {
+	Value    ir.String
+	Metadata metadataContext
+}
+
+// List of metadata attributes supported.
+type metadataContext struct {
```

Comment:
Of course, a "time of assembly" field is necessary here too, so we can remember when a node "entered" the record. This will be set by the Assembler.